config/config.go

```diff
@@ -505,6 +505,8 @@ type HTTPClientConfig struct {
 	ProxyURL URL `yaml:"proxy_url,omitempty"`
 	// TLSConfig to use to connect to the targets.
 	TLSConfig TLSConfig `yaml:"tls_config,omitempty"`
+	// If set, override whether to use HTTP KeepAlive - scraping defaults OFF, remote read/write defaults ON
+	KeepAlive *bool `yaml:"keep_alive,omitempty"`
```

Comment:
I don't think this should be configurable.

Reply:
Maybe that should be a comment to #2498 ?

Reply:
That issue isn't about remote storage

Reply:
That line makes it configurable for scraping too.

Are you saying it should be non-configurable solely for remote storage?  (which would be a change [here](https://github.com/prometheus/prometheus/pull/3171/files#diff-ae8db9d16d8057358e49d694522e7186R55)

Reply:
I think it should be non-configurable generally.

Reply:
OK, so your comment does belong on #2498.

Also you might want to revert this change off master.